internal/resolvers/youtube/resolver.go

```diff
@@ -51,6 +62,26 @@ func New() (resolvers []resolver.CustomURLManager) {
 		return
 	}
 
+	resolvers = append(resolvers, resolver.CustomURLManager{
+		// TODO(jammeh): handle channels where channel/user segment doesn't exist (e.g. https://www.youtube.com/tranceluv)
+		Check: func(url *url.URL) bool {
+			matches, regexErr := regexp.MatchString(`/(user|c(hannel)?)/.+`, url.Path)
+			return utils.IsSubdomainOf(url, "youtube.com") && regexErr == nil && matches
```

Comment:
1. It's worth checking if we can replace `.+` in this regex with something like `\w+` (or `[-\w]`), not exactly sure what are the rules for channel name / ID on youtube.
2. Can we declare the regex as `regexp.Regexp` and then use `(*regexp.Regexp).MatchString()` instead? For the reference, see `check.go` files in Discord resolver or Twitter resolver. I think this would make our usage of regexps more consistent and could also remove `regexErr == nil` from this check in the `return`.

Reply:
1. Will have a look, pretty sure only characters used are alphanumeric + `[_-]`
2. Can do!